x-pack/plugins/apm/public/utils/formatters.ts

```diff
@@ -141,3 +141,88 @@ export function asPercent(
   const decimal = numerator / denominator;
   return numeral(decimal).format('0.0%');
 }
+
+type ByteFormatter = (value: number | null) => string;
+
+function asKilobytes(value: number | null) {
+  if (value === null || isNaN(value)) {
+    return '';
+  }
+  return `${asDecimal(value / 1000)} KB`;
+}
+
+function asMegabytes(value: number | null) {
+  if (value === null || isNaN(value)) {
+    return '';
+  }
+  return `${asDecimal(value / 1e6)} MB`;
+}
+
+function asGigabytes(value: number | null) {
+  if (value === null || isNaN(value)) {
+    return '';
+  }
+  return `${asDecimal(value / 1e9)} GB`;
+}
+
+function asTerabytes(value: number | null) {
+  if (value === null || isNaN(value)) {
+    return '';
+  }
+  return `${asDecimal(value / 1e12)} TB`;
+}
+
+export function asBytes(value: number | null) {
+  if (value === null || isNaN(value)) {
+    return '';
+  }
+  return `${asDecimal(value)} B`;
+}
+
+export function asDynamicBytes(value: number | null) {
+  if (value === null || isNaN(value)) {
```

Comment:
Minor nit:
```ts
export function asDynamicBytes(value: number | null) {
    return getFixedByteFormatter(value)(value)
}
```

(would probably be better to call an non-memoized version of `getFixedByteFormatter`)

Reply:
It's fine to just call the memoized one, don't you think?

Reply:
Or do you think this creates too many cached versions of the functions?